pdns/ixfrdist.cc

```diff
@@ -619,8 +620,7 @@ void handleTCPRequest(int fd, boost::any&) {
     return;
   }
 
-  char buf[4096];
-  // Discard the first 2 bytes (qlen)
+  char buf[65535];
```

Comment:
That's a huge buffer on the stack, perhaps 4k was enough for an `XFR` query? BTW you are shadowing this variable line 710:
```
char buf[2];
```